golint on some pkg/* packages

vdemeester opened the pull request:
I started to work on `golint` the `pkg/*` package 🐸.
- pkg/useragent
- pkg/units
- pkg/ulimit
- pkg/truncindex
- pkg/timeoutconn
- pkg/term
- pkg/tarsum
- pkg/tailfile
- pkg/systemd
- pkg/stringutils
- pkg/stringid
- pkg/streamformatter
- pkg/sockets
- pkg/signals
- pkg/proxy
- pkg/progressreader
- pkg/pools
- pkg/plugins
- pkg/pidfile
- pkg/parsers
- pkg/parsers/filters
- pkg/parsers/kernel
- pkg/parsers/operatingsystem

Few _TODOs_ :
- [X] <del>`pkg/term/winconsole`</del> see comments below.
- [X] Add some package documentation

This PR is part of #14756 🐧. It also covers a bit of #11578.

Signed-off-by: Vincent Demeester vincent@sbr.pm

vdemeester commented:
In ``pkg/ulimit``, golint doesn't really like the const (https://github.com/docker/docker/blob/master/pkg/ulimit/ulimit.go#L27). I'm wondering how to change that though..

aaronlehmann commented on pkg/parsers/filters/parse.go:
```diff
@@ -7,16 +7,17 @@ import (
 type Args map[string][]string
 
-// Parse the argument to the filter flag. Like
+// ParseFlag parse the argument to the filter flag. Like
```
s/parse/parses/

aaronlehmann commented on pkg/parsers/filters/parse.go:
```diff
@@ -7,16 +7,17 @@ import (
 // map is created.
 func ParseFlag(arg string, prev Args) (Args, error) {
-	var filters Args = prev
+	var filters = prev
```
How about `filters := prev` ?

aaronlehmann commented on pkg/parsers/filters/parse.go:
```diff
@@ -36,9 +37,10 @@ func ParseFlag(arg string, prev Args) (Args, error) {
 	return filters, nil
 }
 
+// ErrorBadFormat bad format for filter
```
How about "ErrorBadFormat is an error returned in case of a bad format for a filter"?

runcom replied:
this should also be just `ErrBadFormat` like in stdlib

stevvooe replied:
@runcom Agreed. `ErrBadFormat` is the most idiomatic, unless these are typed constants for a locally defined `Error` type.

aaronlehmann commented on pkg/parsers/filters/parse.go:
```diff
@@ -64,6 +66,8 @@ func FromParam(p string) (Args, error) {
 }
 
+// MatchKVList return true if the values for the specified field maches the ones
+// from the sources.
```
s/return/returns/

aaronlehmann commented on pkg/parsers/filters/parse.go:
```diff
@@ -96,6 +100,7 @@ outer:
 	return true
 }
 
+// Match return true if the values for the specified field matches the source string
```
s/return/returns/

aaronlehmann commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -8,20 +8,21 @@ import (
 }
 
-// Compare two KernelVersionInfo struct.
+// CompareKernelVersion Compare two KernelVersionInfo struct.
```
"CompareKernelVersion compares two KernelVersionInfo structs"

aaronlehmann commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -43,7 +44,8 @@ func CompareKernelVersion(a, b *KernelVersionInfo) int {
 }
 
-func GetKernelVersion() (*KernelVersionInfo, error) {
+// GetKernelVersion get the current kernel version
```
s/get/gets/

aaronlehmann commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -67,7 +69,8 @@ func GetKernelVersion() (*KernelVersionInfo, error) {
 }
 
-func ParseRelease(release string) (*KernelVersionInfo, error) {
+// ParseRelease parse a string and creates a VersionInfo based on it
```
s/parse/parses/

aaronlehmann commented on pkg/parsers/kernel/uname_linux.go:
```diff
@@ -4,6 +4,7 @@ import (
 	"syscall"
 )
 
+// Utsname utsname structure is used to hold information returned by the uname function
```
s/Utsname utsname structure/The Utsname structure/

runcom replied:
agree but w/o "The" as per godoc conv

aaronlehmann commented on pkg/parsers/operatingsystem/operatingsystem_linux.go:
```diff
@@ -14,6 +14,7 @@ var (
 	etcOsRelease = "/etc/os-release"
 )
 
+// GetOperatingSystem get the name of the current operating system
```
s/get/gets/

aaronlehmann commented on pkg/parsers/operatingsystem/operatingsystem_linux.go:
```diff
@@ -26,6 +27,7 @@ func GetOperatingSystem() (string, error) {
 	return "", errors.New("PRETTY_NAME not found")
 }
 
+// IsContainerized are we running inside of a container
```
"IsContainerized returns true if we are running inside of a container"

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -8,6 +8,8 @@ import (
 )
 
+// ParseHost parse the specified address and returns an address that will be used as host
+// Depending of the address specified, will use the defaultTCPAddr or defaultUnixAddr
```
s/parse/parses/
s/as host/as the host/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -36,6 +38,8 @@ func ParseHost(defaultTCPAddr, defaultUnixAddr, addr string) (string, error) {
 }
 
+// ParseUnixAddr parse and validate that the specified address is a valid unix address
+// and returns a formatted unix address, that might use the specified default address
```
s/parse and validate/parses and validates/
s/that might use/that may be used/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -47,6 +51,8 @@ func ParseUnixAddr(addr string, defaultAddr string) (string, error) {
 }
 
+// ParseTCPAddr parse and validate that the specified address is a valid tcp address
+// and returns a formatted tcp address, that might use the specified default address
```
s/parse and validate/parses and validates/
s/that might use/that may be used/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -73,7 +79,7 @@ func ParseTCPAddr(addr string, defaultAddr string) (string, error) {
 }
 
-// Get a repos name and returns the right reposName + tag|digest
+// ParseRepositoryTag Get a repos name and returns the right reposName + tag|digest
```
s/Get/gets/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -93,6 +99,8 @@ func ParseRepositoryTag(repos string) (string, string) {
 }
 
+// PartParser validate and parse the specified string (data) using the specified template
+// e.g. ip:public:private -> 192.168.0.1:80:8000
```
s/validate and parse/validates and parses/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -114,6 +122,7 @@ func PartParser(template, data string) (map[string]string, error) {
 	return out, nil
 }
 
+// ParseKeyValueOpt validate and parse the specified string as a key/value (key=value)
```
s/validate and parse/validates and parses/
s/key\/value/key\/value pair/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -122,6 +131,7 @@ func ParseKeyValueOpt(opt string) (string, string, error) {
 	return strings.TrimSpace(parts[0]), strings.TrimSpace(parts[1]), nil
 }
 
+// ParsePortRange validate and parse the specified string as a port-range (8000-9000)
```
s/validate and parse/validates and parses/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -147,6 +157,7 @@ func ParsePortRange(ports string) (uint64, uint64, error) {
 	return start, end, nil
 }
 
+// ParseLink validate and parse the specified string as a link format (name:alias)
```
s/validate and parse/validates and parses/

aaronlehmann commented on pkg/pidfile/pidfile.go:
```diff
@@ -8,6 +8,7 @@ import (
 	"strconv"
 )
 
+// PidFile file used as a pid
```
I think to keep with the convention of keeping acronyms in uppercase, this should be PIDFile

The comment could read "A PIDFile is a file used to store the process ID of a running process"

aaronlehmann commented on pkg/pidfile/pidfile.go:
```diff
@@ -23,6 +24,7 @@ func checkPidFileAlreadyExists(path string) error {
 	return nil
 }
 
+// New Create a Pidfile using the specified path
```
s/Create/creates/

aaronlehmann commented on pkg/pidfile/pidfile.go:
```diff
@@ -34,6 +36,7 @@ func New(path string) (*PidFile, error) {
 	return &PidFile{path: path}, nil
 }
 
+// Remove remove the PidFile
```
s/remove/removes/

aaronlehmann commented on pkg/plugins/client.go:
```diff
@@ -19,6 +19,7 @@ const (
 	defaultTimeOut  = 30
 )
 
+// NewClient create a new plugin client (http)
```
s/create/creates/

aaronlehmann commented on pkg/plugins/client.go:
```diff
@@ -33,11 +34,13 @@ func NewClient(addr string, tlsConfig tlsconfig.Options) (*Client, error) {
 	return &Client{&http.Client{Transport: tr}, protoAndAddr[1]}, nil
 }
 
+// Client plugin client
```
"Client represents a plugin client"

aaronlehmann commented on pkg/plugins/client.go:
```diff
@@ -33,11 +34,13 @@ func NewClient(addr string, tlsConfig tlsconfig.Options) (*Client, error) {
+	addr string       // http address of the plugin
 }
 
+// Call call the specified method with the specified arguments for the plugin
```
s/call/calls/

aaronlehmann commented on pkg/plugins/discovery.go:
```diff
@@ -12,22 +12,28 @@ import (
 	return LocalRegistry{}
 }
 
+// Plugin return the plugin registered with the given name (or returns an error)
```
s/reutrn/returns/

aaronlehmann commented on pkg/plugins/discovery.go:
```diff
@@ -12,22 +12,28 @@ import (
 	Plugin(name string) (*Plugin, error)
 }
 
+// LocalRegistry registry that is local (using unix socket)
```
s/LocalRegistry/A LocalRegistry is a/

aaronlehmann commented on pkg/plugins/discovery.go:
```diff
@@ -12,22 +12,28 @@ import (
 
+// Registry defines behavior of a registry of plugins
 type Registry interface {
+	// Plugins list all plugins
```
s/list/lists/

aaronlehmann commented on pkg/plugins/discovery.go:
```diff
@@ -12,22 +12,28 @@ import (
 type Registry interface {
+	// Plugins list all plugins
 	Plugins() ([]*Plugin, error)
+	// Plugin return the plugin registered with the given name (or returns an error)
```
s/return/returns/

aaronlehmann commented on pkg/plugins/plugins.go:
```diff
@@ -9,6 +9,7 @@ import (
 )
 
 var (
+	// ErrNotImplements the plugin does not implement the request driver
```
s/ErrNotImplements/ErrNotImplements is returned if/

aaronlehmann commented on pkg/plugins/plugins.go:
```diff
@@ -22,16 +23,18 @@ var (
 	extpointHandlers = make(map[string]func(string, *Client))
 )
 
+// Manifest manifest that list what the plugin implements
```
"A manifest lists what a plugin implements"

aaronlehmann commented on pkg/plugins/plugins.go:
```diff
@@ -22,16 +23,18 @@ var (
 	Implements []string
 }
 
+// Plugin definition of a docker plugin
```
"Plugin is the definition of a docker plugin"

aaronlehmann commented on pkg/plugins/plugins.go:
```diff
@@ -96,6 +99,7 @@ func get(name string) (*Plugin, error) {
 	return pl, nil
 }
 
+// Get return the plugin given the specified name and requested implementation
```
s/return/returns/

aaronlehmann commented on pkg/plugins/plugins.go:
```diff
@@ -110,6 +114,7 @@ func Get(name, imp string) (*Plugin, error) {
 	return nil, ErrNotImplements
 }
 
+// Handle add the specified function to the extpointHandlers
```
s/add/adds/

aaronlehmann commented on pkg/pools/pools.go:
```diff
@@ -18,14 +18,15 @@ import (
+	// BufioReader32KPool Pool which returns bufio.Reader with a 32K buffer
 	BufioReader32KPool *BufioReaderPool
-	// Pool which returns bufio.Writer with a 32K buffer
+	// BufioWriter32KPool Pool which returns bufio.Writer with a 32K buffer
```
For both of the above, s/Pool/is a pool/

aaronlehmann commented on pkg/pools/pools.go:
```diff
@@ -18,14 +18,15 @@ import (
 
 const buffer32K = 32 * 1024
 
+// BufioReaderPool bufio reader that uses sync.Pool
```
s/bufio/is a bufio/

aaronlehmann commented on pkg/pools/pools.go:
```diff
@@ -77,6 +78,7 @@ func (bufPool *BufioReaderPool) NewReadCloserWrapper(buf *bufio.Reader, r io.Rea
 	})
 }
 
+// BufioWriterPool bufio writer that uses sync.Pool
```
s/bufio/is a bufio/

aaronlehmann commented on pkg/progressreader/progressreader.go:
```diff
@@ -7,7 +7,7 @@ import (
 )
 
-// Reader with progress bar
+// Config Reader with progress bar
```
"Config contains the configuration for a Reader with a progress bar"

aaronlehmann commented on pkg/progressreader/progressreader.go:
```diff
@@ -20,6 +20,7 @@ type Config struct {
 	Action     string
 }
 
+// New create a new Config
```
s/create/creates/

aaronlehmann commented on pkg/progressreader/progressreader.go:
```diff
@@ -48,6 +49,7 @@ func (config *Config) Read(p []byte) (n int, err error) {
 	return read, err
 }
 
+// Close close the reader (Config)
```
s/close/closes/

aaronlehmann commented on pkg/proxy/proxy.go:
```diff
@@ -5,18 +5,20 @@ import (
 	"net"
 )
 
+// Proxy defines behavior of a proxy
```
s/behavior/the behavior/

aaronlehmann commented on pkg/proxy/proxy.go:
```diff
@@ -5,18 +5,20 @@ import (
+// Proxy defines behavior of a proxy
 type Proxy interface {
-	// Start forwarding traffic back and forth the front and back-end
+	// Run Start forwarding traffic back and forth the front and back-end
```
s/Start/starts/

aaronlehmann commented on pkg/proxy/proxy.go:
```diff
@@ -5,18 +5,20 @@ import (
 	// addresses.
 	Run()
-	// Stop forwarding traffic and close both ends of the Proxy.
+	// Close Stop forwarding traffic and close both ends of the Proxy.
```
s/Stop/stops/
s/close/closes/

aaronlehmann commented on pkg/proxy/proxy.go:
```diff
@@ -5,18 +5,20 @@ import (
+	// Close Stop forwarding traffic and close both ends of the Proxy.
 	Close()
-	// Return the address on which the proxy is listening.
+	// FrontendAddr Return the address on which the proxy is listening.
```
s/Return/returns/

aaronlehmann commented on pkg/proxy/proxy.go:
```diff
@@ -5,18 +5,20 @@ import (
+	// FrontendAddr Return the address on which the proxy is listening.
 	FrontendAddr() net.Addr
-	// Return the proxied address.
+	// BackendAddr Return the proxied address.
```
s/Return/returns/

aaronlehmann commented on pkg/proxy/stub_proxy.go:
```diff
@@ -4,16 +4,25 @@ import (
 	"net"
 )
 
+// StubProxy a proxy that is a stub (does nothing)
```
s/StubProxy/StubProxy is/

aaronlehmann commented on pkg/proxy/stub_proxy.go:
```diff
@@ -4,16 +4,25 @@ import (
 func (p *StubProxy) FrontendAddr() net.Addr { return p.frontendAddr }
-func (p *StubProxy) BackendAddr() net.Addr  { return p.backendAddr }
 
+// BackendAddr return the backend address
```
s/return/returns/ for both FrontendAddr and BackendAddr

aaronlehmann commented on pkg/proxy/tcp_proxy.go:
```diff
@@ -8,12 +8,14 @@ import (
 	"github.com/Sirupsen/logrus"
 )
 
+// TCPProxy proxy that use TCP address and listener
```
s/proxy/is a proxy/
s/use/uses a/

aaronlehmann commented on pkg/proxy/tcp_proxy.go:
```diff
@@ -72,6 +74,7 @@ func (proxy *TCPProxy) clientLoop(client *net.TCPConn, quit chan bool) {
 	backend.Close()
 }
 
+// Run start forwarding the traffic using TCP.
```
s/start/starts/

aaronlehmann commented on pkg/proxy/tcp_proxy.go:
```diff
@@ -85,6 +88,11 @@ func (proxy *TCPProxy) Run() {
 }
 
-func (proxy *TCPProxy) Close()                 { proxy.listener.Close() }
+// Close stop forwarding the traffic.
```
s/stop/stops/

aaronlehmann commented on pkg/proxy/tcp_proxy.go:
```diff
@@ -85,6 +88,11 @@ func (proxy *TCPProxy) Run() {
 func (proxy *TCPProxy) FrontendAddr() net.Addr { return proxy.frontendAddr }
-func (proxy *TCPProxy) BackendAddr() net.Addr  { return proxy.backendAddr }
+
+// BackendAddr return the tcp proxied address.
```
s/return/returns/ for both FrontendAddr and BackendAddr
s/tcp/TCP/

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -12,8 +12,10 @@ import (
 )
 
 const (
+	// UDPConnTrackTimeout UDP timeout
```
"UDPConnTrackTimeout is the timeout used for UDP connection tracking"

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -12,8 +12,10 @@ import (
+	// UDPConnTrackTimeout UDP timeout
 	UDPConnTrackTimeout = 90 * time.Second
-	UDPBufSize          = 65507
+	// UDPBufSize UDP size for the buffer use to proxy
```
"UDPBufSize is the buffer size used for the UDP proxy"

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -41,6 +43,7 @@ func newConnTrackKey(addr *net.UDPAddr) *connTrackKey {
 
 type connTrackMap map[connTrackKey]*net.UDPConn
 
+// UDPProxy proxy that use UDP address and listener
```
"UDPProxy is a proxy for which handles UDP datagrams"

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -96,6 +100,7 @@ func (proxy *UDPProxy) replyLoop(proxyConn *net.UDPConn, clientAddr *net.UDPAddr
 	}
 }
 
+// Run start forwarding the traffic using UDP.
```
s/start/starts/

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -135,6 +140,7 @@ func (proxy *UDPProxy) Run() {
 	}
 }
 
+// Close stop forwarding the traffic.
```
s/stop/stops/

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -144,8 +150,11 @@ func (proxy *UDPProxy) Close() {
 	}
 }
 
+// FrontendAddr return the udp address on which the proxy is listening.
```
s/return/returns/
s/udp/UDP/

aaronlehmann commented on pkg/proxy/udp_proxy.go:
```diff
@@ -144,8 +150,11 @@ func (proxy *UDPProxy) Close() {
 func (proxy *UDPProxy) FrontendAddr() net.Addr { return proxy.frontendAddr }
-func (proxy *UDPProxy) BackendAddr() net.Addr  { return proxy.backendAddr }
+
+// BackendAddr return the upd proxied address.
```
s/return/returns/
s/upd proxied/proxied UDP/

aaronlehmann commented on pkg/signal/signal.go:
```diff
@@ -5,6 +5,7 @@ import (
 	"os/signal"
 )
 
+// CatchAll catches all signals and relay them to the specified channel
```
s/relay/relays/

aaronlehmann commented on pkg/signal/signal.go:
```diff
@@ -13,6 +14,7 @@ func CatchAll(sigc chan os.Signal) {
 	signal.Notify(sigc, handledSigs...)
 }
 
+// StopCatch stops catching the signals and close the specified channel
```
s/close/closes/

aaronlehmann commented on pkg/signal/signal_linux.go:
```diff
@@ -4,6 +4,7 @@ import (
 	"syscall"
 )
 
+// SignalMap map of Linux signals
```
s/map/is a map/

aaronlehmann commented on pkg/signal/signal_unix.go:
```diff
@@ -8,5 +8,9 @@ import (
 const SIGCHLD = syscall.SIGCHLD
+
+// signal sent to a process when its controlling terminal changes its size
 const SIGWINCH = syscall.SIGWINCH
```
Don't these have to start with "SIGCHLD is..." and so on to make godoc and golint happy?

vdemeester replied:
I thought so but ``golint`` is quite happy about it. It might have something to do with the build flags (my naive guess).

aaronlehmann commented on pkg/signal/trap.go:
```diff
@@ -55,6 +55,7 @@ func Trap(cleanup func()) {
 	}()
 }
 
+// DumpStacks dump the runtime stack
```
s/dump/dumps/

aaronlehmann commented on pkg/sockets/tcp_socket.go:
```diff
@@ -9,7 +9,12 @@ import (
+// and the specified tls configuration. If tlsConfig is set, will encapsulate the
+// tcp listener inside a tls one.
+// The channel passed is used to activate the listenbuffer when the caller is ready
+// to accept connections.
```
Looks good, but please uppercase TLS and TCP in the comment.

aaronlehmann commented on pkg/sockets/tcp_socket.go:
```diff
@@ -21,6 +26,10 @@ func NewTcpSocket(addr string, tlsConfig *tls.Config, activate <-chan struct{})
 }
 
+// ConfigureTCPTransport configure the specified Transport according to the
+// specified proto and addr.
```
s/configure/configures/

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -37,6 +39,7 @@ func (sf *StreamFormatter) FormatStream(str string) []byte {
 	return []byte(str + "\r")
 }
 
+// FormatStatus formate the specified objects using objecs according to the specified format (and id)
```
s/formate/formats/

Should the words "using objecs" be removed?

vdemeester replied:
Oh yeah :blush: :sweat_smile:

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/docker/docker/pkg/jsonmessage"
 )
 
+// StreamFormatter Format a stream using or not json
```
"StreamFormatter formats a stream, optionally using JSON"

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -26,6 +27,7 @@ const streamNewline = "\r\n"
 
 var streamNewlineBytes = []byte(streamNewline)
 
+// FormatStream Format the specified stream
```
s/Format/formats/

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -49,6 +52,7 @@ func (sf *StreamFormatter) FormatStatus(id, format string, a ...interface{}) []b
 	return []byte(str + streamNewline)
 }
 
+// FormatError format the specifed error
```
s/format/formats/

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -63,6 +67,7 @@ func (sf *StreamFormatter) FormatError(err error) []byte {
 	return []byte("Error: " + err.Error() + streamNewline)
 }
 
+// FormatProgress format the specified progress for the specified action
```
"FormatProgress formats progress information for a specified action"

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -86,12 +91,13 @@ func (sf *StreamFormatter) FormatProgress(id, action string, progress *jsonmessa
 }
 
-type StdoutFormater struct {
+// StdoutFormatter streamFormatter that writes in the standard output
```
"StdoutFormatter is a StreamFormatter that writes to the standard output"

aaronlehmann commented on pkg/streamformatter/streamformatter.go:
```diff
@@ -100,12 +106,13 @@ func (sf *StdoutFormater) Write(buf []byte) (int, error) {
 }
 
-type StderrFormater struct {
+// StderrFormatter streamFormatter that writes in the standard error
```
"StdeddFormatter is a StreamFormatter that writes to the standard error"

aaronlehmann commented on pkg/stringid/stringid.go:
```diff
@@ -12,7 +12,7 @@ const shortLen = 12
 var validShortID = regexp.MustCompile("^[a-z0-9]{12}$")
 
-// Determine if an arbitrary string *looks like* a short ID.
+// IsShortID Determine if an arbitrary string *looks like* a short ID.
```
s/Determine/determines/

aaronlehmann commented on pkg/stringutils/stringutils.go:
```diff
@@ -8,7 +8,7 @@ import (
 )
 
-// Generate alpha only random stirng with length n
+// GenerateRandomAlphaOnlyString Generate alpha only random stirng with length n
```
"GenerateRandomAlphaOnlyString generates an alphabetical random string with length n"

aaronlehmann commented on pkg/stringutils/stringutils.go:
```diff
@@ -20,8 +20,8 @@ func GenerateRandomAlphaOnlyString(n int) string {
 
-// Generate Ascii random stirng with length n
-func GenerateRandomAsciiString(n int) string {
+// GenerateRandomASCIIString Generate Ascii random stirng with length n
```
"GenerateRandomAlphaOnlyString generates an ASCII random string with length n"

aaronlehmann commented on pkg/stringutils/stringutils.go:
```diff
@@ -40,7 +40,7 @@ func Truncate(s string, maxlen int) string {
 }
 
-// Test wheather a string is contained in a slice of strings or not.
+// InSlice Test wheather a string is contained in a slice of strings or not.
```
s/Test wheather/tests whether/

aaronlehmann commented on pkg/stringutils/stringutils.go:
```diff
@@ -73,8 +73,8 @@ func quote(word string, buf *bytes.Buffer) {
 
-// Take a list of strings and escape them so they will be handled right
-// when passed as arguments to an program via a shell
+// ShellQuoteArguments Take a list of strings and escape them so they will be
```
s/Take/takes
s/escape/escapes/

aaronlehmann commented on pkg/systemd/booted.go:
```diff
@@ -4,7 +4,7 @@ import (
 )
 
-// Conversion to Go of systemd's sd_booted()
+// SdBooted Conversion to Go of systemd's sd_booted()
```
"SdBooted is systemd's sd_booted() ported to Go"

aaronlehmann commented on pkg/systemd/sd_notify.go:
```diff
@@ -6,9 +6,10 @@ import (
 )
 
-var SdNotifyNoSocket = errors.New("No socket")
+// ErrSdNotifyNoSocket no socket was specified
```
"ErrSdNotifyNoSocket is an error returned if no socket was specified"

aaronlehmann commented on pkg/systemd/sd_notify.go:
```diff
@@ -6,9 +6,10 @@ import (
+var ErrSdNotifyNoSocket = errors.New("No socket")
 
-// Send a message to the init daemon. It is common to ignore the error.
+// SdNotify Send a message to the init daemon. It is common to ignore the error.
```
"SdNotify sends a message to the init daemon. It is common to ignore the return value."

aaronlehmann commented on pkg/tailfile/tailfile.go:
```diff
@@ -9,6 +9,8 @@ import (
 
 var eol = []byte("\n")
+
+// ErrNonPositiveLinesNumber the lines number was negative
```
"ErrNonPositiveLinesNumber is an error returned if the lines number was negative"

aaronlehmann commented on pkg/tailfile/tailfile.go:
```diff
@@ -9,6 +9,8 @@ import (
 var eol = []byte("\n")
+
+// ErrNonPositiveLinesNumber the lines number was negative
 var ErrNonPositiveLinesNumber = errors.New("Lines number must be positive")
```
Not related to your changes, but text would be clearer as "The number of lines to extract from the file must be positive"

vdemeester replied:
agree, I'll update this too.

aaronlehmann commented on pkg/tarsum/builder_context.go:
```diff
@@ -1,9 +1,10 @@
-// there was concern about adding this method to TarSum itself so instead
-// it is being added just to "BuilderContext" which will then only be used
-// during the .dockerignore file processing - see builder/evaluator.go
+// BuilderContext This interface extends TarSum by adding the Remove method.
```
"BuilderContext is an interface extending TarSum by adding the Remove method."

aaronlehmann commented on pkg/ulimit/ulimit.go:
```diff
@@ -24,43 +25,44 @@ const (
-	"rttime":     RLIMIT_RTTIME,
-	"sigpending": RLIMIT_SIGPENDING,
-	"stack":      RLIMIT_STACK,
+	//"as":         rlimitAs, // Disbaled since this doesn't seem usable with the way Docker inits a container.
```
The comment has a typo - should be "Disabled"

aaronlehmann commented on pkg/ulimit/ulimit.go:
```diff
@@ -24,43 +25,44 @@ const (
+	rlimitRtprio     = 14
+	rlimitRttime     = 15
+	rlimitSigpending = 11
+	rlimitStack      = 3
```
Nice, it looks a lot cleaner to avoid these all-caps names.

vdemeester replied:
yeah, and ``golint`` doesn't like the ALL_CAPS anyway :blush:.

aaronlehmann commented on pkg/tarsum/fileinfosums.go:
```diff
@@ -2,7 +2,8 @@ package tarsum
 
-// This info will be accessed through interface so the actual name and sum cannot be medled with
+// FileInfoSumInterface This info will be accessed through interface
+// so the actual name and sum cannot be medled with
```
"FileInfoSumInterface provides an interface for accessing file checksum information within a tar file. This info is accessed through interface so the actual name and sum cannot be medled with."

vdemeester replied:
<3

aaronlehmann commented on pkg/tarsum/fileinfosums.go:
```diff
@@ -28,6 +29,7 @@ func (fis fileInfoSum) Pos() int64 {
 	return fis.pos
 }
 
+// FileInfoSums list of FileInfoSumInterface
```
"The FileInfoSums type provides a list of FileInfoSumInterfaces"

aaronlehmann commented on pkg/tarsum/fileinfosums.go:
```diff
@@ -51,6 +53,7 @@ func (fis FileInfoSums) GetAllFile(name string) FileInfoSums {
 	return f
 }
 
+// GetDuplicatePaths returns a FileInfoSums with all duplicates paths
```
s/duplicates/duplicate/

aaronlehmann commented on pkg/tarsum/fileinfosums.go:
```diff
@@ -64,17 +67,23 @@ func (fis FileInfoSums) GetDuplicatePaths() (dups FileInfoSums) {
 }
 
-func (fis FileInfoSums) Len() int      { return len(fis) }
+// Len size of the FileInfoSums
```
s/Len size/Len returns the size/

aaronlehmann commented on pkg/tarsum/fileinfosums.go:
```diff
@@ -64,17 +67,23 @@ func (fis FileInfoSums) GetDuplicatePaths() (dups FileInfoSums) {
+// Len size of the FileInfoSums
+func (fis FileInfoSums) Len() int { return len(fis) }
+
+// Swap swap two FileInfoSum of the FileInfoSums
```
"Swap swaps two FileInfoSum values in a FileInfoSums list"

aaronlehmann commented on pkg/tarsum/fileinfosums.go:
```diff
@@ -64,17 +67,23 @@ func (fis FileInfoSums) GetDuplicatePaths() (dups FileInfoSums) {
 	sort.Sort(byName{fis})
 }
 
+// SortBySums sort FileInfoSums content by sums
```
s/sort/sorts/ for all three of the above

aaronlehmann commented on pkg/tarsum/tarsum.go:
```diff
@@ -30,7 +30,8 @@ func NewTarSum(r io.Reader, dc bool, v Version) (TarSum, error) {
 }
 
-// Create a new TarSum, providing a THash to use rather than the DefaultTHash
+// NewTarSumHash Create a new TarSum, providing a THash to use rather than
```
s/Create/creates/

aaronlehmann commented on pkg/tarsum/tarsum.go:
```diff
@@ -41,7 +42,7 @@ func NewTarSumHash(r io.Reader, dc bool, v Version, tHash THash) (TarSum, error)
 }
 
-// Create a new TarSum using the provided TarSum version+hash label.
+// NewTarSumForLabel Create a new TarSum using the provided TarSum version+hash label.
```
s/Create/creates/

aaronlehmann commented on pkg/tarsum/tarsum.go:
```diff
@@ -104,13 +105,13 @@ func (ts tarSum) Version() Version {
 }
 
-// A hash.Hash type generator and its name
+// THash A hash.Hash type generator and its name
```
s/THash A/THash provides a/

aaronlehmann commented on pkg/tarsum/tarsum.go:
```diff
@@ -104,13 +105,13 @@ func (ts tarSum) Version() Version {
 }
 
-// Convenience method for creating a THash
+// NewTHash Convenience method for creating a THash
```
s/Convenience/is a convenience/

aaronlehmann commented on pkg/tarsum/versioning.go:
```diff
@@ -8,7 +8,7 @@ import (
 )
 
-// versioning of the TarSum algorithm
+// Version versioning of the TarSum algorithm
```
s/Version/Version is used for/

aaronlehmann commented on pkg/tarsum/versioning.go:
```diff
@@ -17,7 +17,7 @@ type Version int
 	Version0 Version = iota
 	Version1
-	// NOTE: this variable will be either the latest or an unsettled next-version of the TarSum calculation
+	// VersionDev (NOTE) this variable will be either the latest or an unsettled next-version of the TarSum calculation
```
This won't look right in godoc, and the comment doesn't really make sense because it says "variable" but it's describing a constant.

aaronlehmann commented on pkg/tarsum/versioning.go:
```diff
@@ -33,7 +33,7 @@ func VersionLabelForChecksum(checksum string) string {
 }
 
-// Get a list of all known tarsum Version
+// GetVersions Get a list of all known tarsum Version
```
s/Get/gets/
s/Version/versions/

aaronlehmann commented on pkg/term/term.go:
```diff
@@ -12,13 +12,16 @@ import (
 )
 
 var (
+	// ErrInvalidState the state of the terminal is invalid
```
s/the/is returned if the/

aaronlehmann commented on pkg/term/term.go:
```diff
@@ -26,10 +29,12 @@ type Winsize struct {
 	return os.Stdin, os.Stdout, os.Stderr
 }
 
+// GetFdInfo returns the file description information of the given input (eg a file)
```
"GetFdInfo returns the file descriptor for an os.File, and indicates whether the file represents a terminal"

aaronlehmann commented on pkg/term/term.go:
```diff
@@ -50,6 +56,7 @@ func GetWinsize(fd uintptr) (*Winsize, error) {
 	return ws, err
 }
 
+// SetWinsize try to set the specified window size based the specified file descriptor
```
s/try/tries/
s/based/for/

aaronlehmann commented on pkg/term/term.go:
```diff
@@ -77,6 +84,7 @@ func RestoreTerminal(fd uintptr, state *State) error {
 	return nil
 }
 
+// SaveState Save the state of the terminal connected to the given file descriptor
```
s/Save/saves/

aaronlehmann commented on pkg/term/term.go:
```diff
@@ -86,6 +94,8 @@ func SaveState(fd uintptr) (*State, error) {
 }
 
+// DisableEcho disable echo of the terminal connected to the given file descriptor
+// according to the given state
```
"DisableEcho applies the specified state to the terminal connected to the file descriptor, with echo disabled"

aaronlehmann commented on pkg/term/term.go:
```diff
@@ -97,6 +107,8 @@ func DisableEcho(fd uintptr, state *State) error {
 }
 
+// SetRawTerminal makes the terminal connected to the given file descriptor into
+// raw mode and returns the previous state
```
s/makes/puts/

aaronlehmann commented on pkg/timeoutconn/timeoutconn.go:
```diff
@@ -5,11 +5,13 @@ import (
 	"time"
 )
 
+// New create a net.Conn with a timeout for every Read or Write operation
```
s/create/creates/

It appears to only apply the timeout for Read operations

vdemeester replied:
Yes, I added a ``FIXME`` comment on the ``conn`` struct (just below) because the comment was wrong. Anyway I'll update that.

aaronlehmann commented on pkg/truncindex/truncindex.go:
```diff
@@ -10,7 +10,9 @@ import (
+	// ErrEmptyPrefix the prefix was empty
+	ErrEmptyPrefix = errors.New("Prefix can't be empty")
+	// ErrAmbiguousPrefix the prefix was ambiguous (multiple ids for the prefix)
 	ErrAmbiguousPrefix = errors.New("Multiple IDs found with provided prefix")
```
s/the prefix/is an error returned if the prefix/

aaronlehmann commented on pkg/ulimit/ulimit.go:
```diff
@@ -6,13 +6,14 @@ import (
 )
 
-// Human friendly version of Rlimit
+// Ulimit Human friendly version of Rlimit
```
s/Human friendly/is a human friendly/

aaronlehmann commented on pkg/ulimit/ulimit.go:
```diff
@@ -24,43 +25,44 @@ const (
+	"stack":      rlimitStack,
 }
 
+// Parse parse and return a Ulimit from the specified string.
```
s/parse and return/parses and returns/

aaronlehmann commented on pkg/useragent/useragent.go:
```diff
@@ -3,15 +3,10 @@
-)
-
-// VersionInfo is used to model UserAgent versions.
+// VersionInfo used to model UserAgent versions.
```
I think we should keep the "is".

aaronlehmann commented on pkg/useragent/useragent.go:
```diff
@@ -30,7 +25,7 @@ func (vi *VersionInfo) isValid() bool {
 }
 
-// Convert versions to a string and append the string to the string base.
+// AppendVersions Convert versions to a string and append the string to the string base.
```
s/Convert/converts/
s/append/appends/

stevvooe commented on pkg/parsers/filters/parse.go:
```diff
@@ -7,16 +7,17 @@ import (
 	"strings"
 )
 
+// Args arguments as map key:{array}
```
This should probably give more about the intent of this type declaration. Why not just use `map[string][]string` directly? What kind of "args" are these?

stevvooe commented on pkg/parsers/filters/parse.go:
```diff
@@ -36,9 +37,10 @@ func ParseFlag(arg string, prev Args) (Args, error) {
 var ErrorBadFormat = errors.New("bad format of filter (expected name=value)")
 
-// packs the Args into an string for easy transport from client to server
+// ToParam packs the Args into an string for easy transport from client to server
```
Should this be plural?

vdemeester replied:
I'm not sure it should be plural, as it returns only one string, don't you think ? :)

stevvooe commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -8,20 +8,21 @@ import (
 	Kernel int
 	Major  int
 	Minor  int
 	Flavor string
```
Need to document the individual fields to pass `golint`.

vdemeester replied:
hum ``golint`` did not say a thing about that, but I'll document them anyway :wink:.

stevvooe commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -8,20 +8,21 @@ import (
+// CompareKernelVersion Compare two KernelVersionInfo struct.
 // Returns -1 if a < b, 0 if a == b, 1 it a > b
-func CompareKernelVersion(a, b *KernelVersionInfo) int {
+func CompareKernelVersion(a, b *VersionInfo) int {
```
No need for pointer type here.

stevvooe commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -43,7 +44,8 @@ func CompareKernelVersion(a, b *KernelVersionInfo) int {
 
-func GetKernelVersion() (*KernelVersionInfo, error) {
+// GetKernelVersion get the current kernel version
+func GetKernelVersion() (*VersionInfo, error) {
```
No need for pointer type here.

vdemeester replied:
Few lines later we return a nil (``return nil, err``), I think that's why the pointer was there.

stevvooe commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -67,7 +69,8 @@ func GetKernelVersion() (*KernelVersionInfo, error) {
 
-func ParseRelease(release string) (*KernelVersionInfo, error) {
+// ParseRelease parse a string and creates a VersionInfo based on it
+func ParseRelease(release string) (*VersionInfo, error) {
```
No need for pointer type here.

vdemeester replied:
Same as above :)

stevvooe commented on pkg/parsers/kernel/uname_linux.go:
```diff
@@ -4,6 +4,7 @@ import (
 )
 
+// Utsname utsname structure is used to hold information returned by the uname function
 type Utsname syscall.Utsname
```
s/`Utsname`/`UTSName`/.

This is an initialism.

stevvooe replied:
oO. I guess they didn't follow this in `syscall`...

vdemeester replied:
Yeah, I *aligned* to the ``syscall.Utsname`` so I did not use ``UTS…``.
But ``type Utsname`` is not used anywhere in docker codebase, the ``uname()`` function returns the ``syscall.Utsname``, so it feels like this type doesn't even need to be *defined*.

vdemeester replied:
Nevermind, it's needed for cross platform I guess :sweat_smile:.

stevvooe commented on pkg/signal/signal_unix.go:
```diff
@@ -8,5 +8,9 @@ import (
 // invalid signals so they don't get handled)
+
+// signal sent to a process when a child process terminates, is interrupted, or resumes after being interrupted.
 const SIGCHLD = syscall.SIGCHLD
```
Could anyone provide some insight on these redefinitions?

vdemeester replied:
It's used in ``api/client/start.go`` to forward signals. I think it's to be able to use this signal on all platform (see https://github.com/docker/docker/blob/master/pkg/signal/signal_windows.go#L11) without doing a *condition*.

stevvooe commented:
@vdemeester This is a good start. I've added some "erroneous" comments around some style issues which might be out of scope for this PR. Feel free to ignore them.

The only feedback I have is to make sure the comments provide more insight that what the function signature provides. Most users come to the godoc to understand why. I understand this can be hard in a sprawling code base, but it is okay to leave a small TODO or question near these comments, indicating more study needs to be done to get to the bottom of this.

Thank you for putting on your archaeologists hat!

vdemeester commented:
@aaronlehmann Thanks a lot for all the typos I've made :blush: :wink:.

@stevvooe Yeah I totally agree on the comments, it has to provide more information than just the name of the function in human readable form. It's not always easy (most on the "why we did that" because there is some history I missed). Overall it's a great exercice to do.

aaronlehmann commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -8,20 +8,21 @@ import (
-	Major  int
-	Minor  int
-	Flavor string
+// VersionInfo holds informations about the kernel.
```
Might have missed this before - should be "information"

aaronlehmann commented on pkg/parsers/kernel/kernel.go:
```diff
@@ -8,20 +8,21 @@ import (
 }
 
-// Compare two KernelVersionInfo struct.
+// CompareKernelVersion compares two KernelVersionInfo struct.
```
s/struct/structs/

vdemeester replied:
Oh and I missed ``KernelVersionInfo`` too :sweat:.

aaronlehmann commented on pkg/parsers/kernel/kernel_windows.go:
```diff
@@ -6,18 +6,20 @@ import (
-	major int
-	minor int
-	build int
+// VersionInfo holds informations about the kernel.
```
s/informations/information/

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -37,6 +39,8 @@ func ParseHost(defaultTCPAddr, defaultUnixAddr, addr string) (string, error) {
 }
 
+// ParseUnixAddr parses and validates that the specified address is a valid unix address
+// and returns a formatted unix address, that may be used the specified default address
```
The last part of this is a bit difficult to understand. I think I mistakenly gave you a bad recommendation here. Maybe "ParseUnixAddr parses and validates that the specified address is a valid UNIX socket address. It returns a formatted UNIX socket address, either using the address parsed from addr, or the contents of defaultAddr if addr is a blank string".

Totally unrelated to this PR, but ParseHost has a silly line:

    addr = fmt.Sprintf("%s", defaultTCPAddr)

vdemeester replied:
:heart_eyes_cat:

aaronlehmann commented on pkg/parsers/parsers.go:
```diff
@@ -48,6 +52,8 @@ func ParseUnixAddr(addr string, defaultAddr string) (string, error) {
 }
 
+// ParseTCPAddr parses and validates that the specified address is a valid tcp address
+// and returns a formatted tcp address, that may be used the specified default address
```
See comment above about ParseUnixAddr

aaronlehmann commented on pkg/plugins/client.go:
```diff
@@ -33,11 +34,14 @@ func NewClient(addr string, tlsConfig tlsconfig.Options) (*Client, error) {
 }
 
+// Call calls the specified method with the specified arguments for the plugin.
+// It will retry for 30 seconds if failures when calling.
```
s/if failures when calling/if a failure occurs/

aaronlehmann commented on pkg/pools/pools.go:
```diff
@@ -77,6 +78,7 @@ func (bufPool *BufioReaderPool) NewReadCloserWrapper(buf *bufio.Reader, r io.Rea
 	})
 }
 
+// BufioWriterPool bufio writer that uses sync.Pool.
```
s/BufioWriterPool/BufioWriterPool is a/

aaronlehmann commented on pkg/signal/signal_linux.go:
```diff
@@ -4,6 +4,7 @@ import (
 	"syscall"
 )
 
+// SignalMap is map of Linux signals.
```
s/is map/is a map/

aaronlehmann commented on pkg/units/size.go:
```diff
@@ -49,17 +49,19 @@ func CustomSize(format string, size float64, base float64, _map []string) string
 }
 
+// BytesSize returns a human-readable size bytes, kibibytes,
+// mebibytes, gibibytes, or tebibytes (eg. "44kiB", "17MiB").
```
s/size bytes/size in bytes/

aaronlehmann commented:
Thanks so much for taking the time to incorporate my feedback. The new diff looks really good. I pointed out a few minor things but it's almost ready to go.

jhowardmsft commented:
LGTM from a Windows perspective. FYI, https://github.com/docker/docker/pull/14838 will change the winconsole stuff, so I would hold on that for a separate PR.

vdemeester commented:
@jhowardmsft sounds good to me for ``winconsole`` :)
@aaronlehmann np :wink:, thans so much for taking the time to make such feedback !!

I'm wondering something, which one is better between :

```go
type VersionInfo struct {
	Kernel int    // Version of the kernel (e.g. 4.1.2-generic -> 4)
	Major  int    // Major part of the kernel version (e.g. 4.1.2-generic -> 1)
	Minor  int    // Minor part of the kernel version (e.g. 4.1.2-generic -> 2)
	Flavor string // Flavor of the kernel version (e.g. 4.1.2-generic -> generic)
}
```

and:

```go
type VersionInfo struct {
	// Version of the kernel (e.g. 4.1.2-generic -> 4)
	Kernel int
	// Major part of the kernel version (e.g. 4.1.2-generic -> 1)
	Major int
	// Minor part of the kernel version (e.g. 4.1.2-generic -> 2)
	Minor int
	// Flavor of the kernel version (e.g. 4.1.2-generic -> generic)
	Flavor string
}
```

stevvooe commented on pkg/proxy/tcp_proxy.go:
```diff
@@ -8,12 +8,14 @@ import (
 	"github.com/Sirupsen/logrus"
 )
 
+// TCPProxy is a proxy for TCP connections.
```
What is the role of a `TCPProxy`? When should it be used? As a resource, does it require management?

vdemeester replied:
I'm tempted to say something like on ``TCPListener`` (http://golang.org/src/net/tcpsock_posix.go?s=7230:7268#L221) as it's mostly meant to be used as ``proxy.NewProxy`` and the function will figure out if it should use ``TCP`` or ``UDP``.

> As a resource, does it require management?

Not sure at all about that :sweat:.

stevvooe commented on pkg/proxy/proxy.go:
```diff
@@ -5,18 +5,20 @@ import (
 	"net"
 )
 
+// Proxy defines the behavior of a proxy.
```
What is that behavior? How does one use it?

vdemeester replied:
I've tried to complete it a bit based on how I saw it use for now : https://github.com/docker/libnetwork/blob/master/portmapper/proxy.go#L43. Not sure that's enough though :blush:.

stevvooe replied:
This is a little a tough to track down. After some discussion with @icecrime, we thought it might be better to overlook some inaccurate comments in the interest of enabling golint across the board. A quick glance at this PR with those goggles shows no problems with this PR as is.

We can go back and do the archaeological work as we make changes to the code base in the future.

vdemeester replied:
Ok :wink:

vdemeester commented:
Rebased and added some package documentation.

vdemeester committed:
Lint on pkg/* packages

- pkg/useragent
- pkg/units
- pkg/ulimit
- pkg/truncindex
- pkg/timeoutconn
- pkg/term
- pkg/tarsum
- pkg/tailfile
- pkg/systemd
- pkg/stringutils
- pkg/stringid
- pkg/streamformatter
- pkg/sockets
- pkg/signal
- pkg/proxy
- pkg/progressreader
- pkg/pools
- pkg/plugins
- pkg/pidfile
- pkg/parsers
- pkg/parsers/filters
- pkg/parsers/kernel
- pkg/parsers/operatingsystem

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

jfrazelle commented:
needs rebase sorry

jfrazelle commented:
feel free to ping me after for merge

vdemeester commented:
@jfrazelle rebased, let's hope janky is a little happier this time :stuck_out_tongue_closed_eyes:.

jfrazelle commented:
LGTM effing dev/urandom errors

vdemeester commented:
:tada:
